### PR TITLE
Issue 6484

### LINKED ISSUE #6484: Registry deployment config needs to set http secret
Upstream documentation advocates setting an [http secret as a must](https://github.com/docker/distribution/blob/master/docs/configuration.md#http) for a cluster of registries behind a load balancer. It coordinate uploads.

Exporting it as an environment variable `REGISTRY_HTTP_SECRET` during `oadm registry` should be sufficient.

By setting it, we'll also prevent similar warnings in a pod's log:

    No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable.

### PR DESCRIPTION
**Pull request by legionus**
This pull request should fix #6484. First commit is part of my previous PR #6653.
@miminar, @kargakis please review.

### CODE REVIEW

**Comment by legionus** [comment]
Fixed.

File: pkg/cmd/admin/registry/registry.go [hunk 1]
```diff
@@ -230,6 +237,17 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
...
+		if len(ports) > 0 {
+			env["REGISTRY_HTTP_ADDR"] = fmt.Sprintf(":%d", ports[0].ContainerPort)
+			env["REGISTRY_HTTP_NET"] = "tcp"
+		}
+
+		var secretBytes [randomSecretSize]byte
+		if _, err := cryptorand.Read(secretBytes[:]); err != nil {
+			return fmt.Errorf("registry does not exist; could not generate random bytes for HTTP secret: %v", err)
+		}
+		env["REGISTRY_HTTP_SECRET"] = fmt.Sprintf("%x", secretBytes)
```
Comment (miminar) [inline pkg/cmd/admin/registry/registry.go]: Just a nitpick. I'd use base64 encoding which will produce much shorter string.

Comment (legionus) [inline pkg/cmd/admin/registry/registry.go]: Fixed too

**Comment by miminar** [comment]
Nice! LGTM

**Comment by miminar** [comment]
[test]

**Comment by legionus** [comment]
[test]

**Comment by mfojtik** [comment]
nice, LGTM, lets waif to test to pass

File: pkg/cmd/admin/registry/registry.go [hunk 2]
```diff
@@ -254,6 +259,12 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
 		livenessProbe := generateLivenessProbeConfig(healthzPort)
 		readinessProbe := generateReadinessProbeConfig(healthzPort)
 
+		var secretBytes [randomSecretSize]byte
+		if _, err := cryptorand.Read(secretBytes[:]); err != nil {
```
Comment (liggitt) [inline pkg/cmd/admin/registry/registry.go]: Make a sized slice, rather than an array you have you slice everywhere you use it

**Comment by legionus** [comment]
Merge ?

**Comment by kargakis** [comment]
LGTM [merge]

**Pull request merged**